arcgis-ios-sdk-samples/Maps/Display drawing status/MapViewDrawStatusViewController.swift

```diff
@@ -45,22 +47,15 @@ class MapViewDrawStatusViewController: UIViewController {
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
-        //add observer for drawStatus on mapView
-        //so we can show/hide an indicator when the status change
-        mapView.addObserver(self, forKeyPath: #keyPath(AGSGeoView.drawStatus), options: .initial, context: nil)
-    }
-    
-    override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
-        DispatchQueue.main.async { [weak self] in
-            guard let strongSelf = self else { return }
-            strongSelf.activityIndicatorView.isHidden = strongSelf.mapView.drawStatus == .completed
+        drawStatusObservation = mapView.observe(\.drawStatus, options: .initial) { [weak self] (mapView, _) in
+            DispatchQueue.main.async {
+                self?.activityIndicatorView.isHidden = mapView.drawStatus == .completed
+            }
         }
     }
     
     override func viewDidDisappear(_ animated: Bool) {
         super.viewDidDisappear(animated)
-        
-        mapView.removeObserver(self, forKeyPath: #keyPath(AGSGeoView.drawStatus))
+        drawStatusObservation = nil
```

Comment:
Shouldn't you call `invalidate()` before setting to `nil`? In iOS 11 this isn't necessary but I think it's still required for iOS 10.

Reply:
I don't think so. From the documentation:

> when the returned NSKeyValueObservation is deinited or invalidated, it will stop observing

And from [the implementation](https://github.com/apple/swift/blob/3f859a3ffd6ee22c765449b773b84fa767a4118f/stdlib/public/SDK/Foundation/NSObject.swift#L159):

> invalidate() will be called automatically when an NSKeyValueObservation is deinited

Reply:
Given our discussion elsewhere, do you think this should include an `invalidate()` for as long as we support iOS 10?